plugins/polio/api/vaccines/supply_chain.py

```diff
@@ -133,16 +130,32 @@ class NestedVaccinePreAlertSerializerForPatch(NestedVaccinePreAlertSerializerFor
     doses_per_vial = serializers.IntegerField(required=False, read_only=True)
     vials_shipped = serializers.IntegerField(required=False, read_only=True)
     document = serializers.FileField(required=False)
+    can_edit = serializers.SerializerMethodField()
 
     class Meta(NestedVaccinePreAlertSerializerForPost.Meta):
-        fields = NestedVaccinePreAlertSerializerForPost.Meta.fields + ["id"]
+        fields = NestedVaccinePreAlertSerializerForPost.Meta.fields + ["id", "can_edit"]
 
     def validate(self, attrs: Any) -> Any:
         # at least one of the other fields must be present
         if not any(key in attrs.keys() for key in NestedVaccinePreAlertSerializerForPost.Meta.fields):
             raise serializers.ValidationError("At least one of the fields must be present.")
+
+        # Check if user has edit permission, if not raise 403
+        if not self.get_can_edit(VaccinePreAlert.objects.get(id=attrs["id"])):
```

Comment:
If I'm not mistaken, you should be able to access `validated_data` somehow in the `validate` method.

This could avoid recomputing `get_can_edit()` with a new DB call.

E.g.

```python
if not validated_data.get("can_edit"):
```